File: 프로그래머스/unrated/136798. 기사단원의 무기/기사단원의 무기.java
class Solution {
    public int solution(int number, int limit, int power) {
        int answer = 1;
		if(number == 1) {
			return 1;
		}
		
		for(int i = 2; i <= number; ++i) {
			int numOfDivisor = 0;
			for(int j = 1; j < Math.sqrt(i); j++) {
				if(i % j == 0) {
					numOfDivisor += 2;
				}
			}
			
			if(isInteger(Math.sqrt(i))) {
				numOfDivisor++;
			}
			
			answer += numOfDivisor > limit ? power : numOfDivisor;
		}
        return answer;
    }
    
    public boolean isInteger(double num) {
		return num == (int)num;
	}
}

File: 프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.js
function solution(number, limit, power) {
    let result = 0;
    for (let i = 1; i <= number; i++){
        let count = 0;
        for (let j = 1; j <= parseInt(Math.sqrt(i)); j++){
            if (i % j === 0) count++;
            if (i % j === 0 && j**2 != i) count++;
        }
        result += count > limit ? power : count;
    }
    return result;
}